server/src-lib/Hasura/GraphQL/Schema.hs

```diff
@@ -1,961 +1,576 @@
-
-    computedFieldReqTypes = catMaybes $
-      maybe [] (map mkComputedFieldRequiredTypes . getComputedFields) selFldsM
-
-    computedFieldFuncArgsInps = map (TIInpObj . fst) computedFieldReqTypes
-    computedFieldFuncArgScalars = Set.fromList $ concatMap snd computedFieldReqTypes
-
-makeFieldMap :: [(a, ObjFldInfo)] -> Map.HashMap G.Name (a, ObjFldInfo)
-makeFieldMap = mapFromL (_fiName . snd)
-
--- see Note [Split schema generation (TODO)]
-getMutationRootFieldsRole
-  :: QualifiedTable
-  -> Maybe (PrimaryKey PGColumnInfo)
-  -> [ConstraintName]
-  -> FieldInfoMap FieldInfo
-  -> Maybe ([T.Text], Bool) -- insert perm
-  -> Maybe (AnnBoolExpPartialSQL, Maybe Int, [T.Text], Bool) -- select filter
-  -> Maybe ([PGColumnInfo], PreSetColsPartial, AnnBoolExpPartialSQL, Maybe AnnBoolExpPartialSQL, [T.Text]) -- update filter
-  -> Maybe (AnnBoolExpPartialSQL, [T.Text]) -- delete filter
-  -> Maybe ViewInfo
-  -> TableConfig -- custom config
-  -> MutationRootFieldMap
-getMutationRootFieldsRole tn primaryKey constraints fields insM
-                           selM updM delM viM tableConfig =
-    makeFieldMap $ catMaybes
-        [ mutHelper viIsInsertable getInsDet insM
-        , onlyIfSelectPermExist $ mutHelper viIsInsertable getInsOneDet insM
-        , mutHelper viIsUpdatable getUpdDet updM
-        , onlyIfSelectPermExist $ mutHelper viIsUpdatable getUpdByPkDet $ (,) <$> updM <*> primaryKey
-        , mutHelper viIsDeletable getDelDet delM
-        , onlyIfSelectPermExist $ mutHelper viIsDeletable getDelByPkDet $ (,) <$> delM <*> primaryKey
+  functionSelectExpParsers <- for allFunctions \function -> do
+    let targetTable = fiReturnType function
+        functionName = fiName function
+    selectPerms <- tableSelectPermissions targetTable
+    for selectPerms \perms -> do
+      displayName <- qualifiedObjectToName functionName
+      let functionDesc = G.Description $ "execute function " <> functionName <<> " which returns " <>> targetTable
+          aggName = displayName <> $$(G.litName "_aggregate")
+          aggDesc = G.Description $ "execute function " <> functionName <<> " and query aggregates on result of table type " <>> targetTable
+      catMaybes <$> sequenceA
+        [ requiredFieldParser (RFDB . QDBSimple)      $ selectFunction          function displayName (Just functionDesc) perms
+        , mapMaybeFieldParser (RFDB . QDBAggregation) $ selectFunctionAggregate function aggName     (Just aggDesc)      perms
         ]
+  actionParsers <- for allActions $ \actionInfo ->
+    case _adType (_aiDefinition actionInfo) of
+      ActionMutation ActionSynchronous -> pure Nothing
+      ActionMutation ActionAsynchronous ->
+        fmap (fmap (RFAction . AQAsync)) <$> actionAsyncQuery actionInfo
+      ActionQuery ->
+        fmap (fmap (RFAction . AQQuery)) <$> actionExecute nonObjectCustomTypes actionInfo
+  pure $ (concat . catMaybes) (tableSelectExpParsers <> functionSelectExpParsers <> toRemoteFieldParser allRemotes)
```

Comment:
What I have on the [blocker](https://github.com/hasura/graphql-engine/pull/4111#issuecomment-677933260) discovered by @jberryman is that if you comment out `tableSelectionExpParsers` *and* comment out `++ maybeToList (P.parserType <$> mutationP)` in `queryWithIntrospectionHelper`, there are no conflicting types. So I think there is a third entry that is causing a conflict somehow. @codingkarthik, do remote relationships get inserted into mutations as well? How is the `order_by` of remote relationships built up?

Reply:
Remote relationships do not get inserted into mutations, but a mutation's return output can contain a remote relationship field. And `order_by` with remote relationships is not supported